### ISSUE TITLE
Low complexity filter and contaminant removal

### ISSUE THREAD

**Issue by martin-hartmann** [OpenGene/fastp#332]
Thanks for a super nice tool. I wonder, if the following to functionalities could be included into fastp.

1) A low complexity filter using methods such as Entropy or Dust. The current filter does not work well on tandem repeats and similar type of low complexity sequences.

2) Removal of contaminants like PhiX. I always see some level of phix remaining in the files that come from the sequencing provider and a functionality in fastp would remove this in the same go as the other steps performed by the tool. Separate mapping e.g. via Bowtie2 is quite slow.

Thanks for considering these options.
Kind regards,
Martin

**Comment by hermidalc** [comment]
Yep I agree, would be nice just saw this when thinking fastp could remove low complexity tandem repeats

**Comment by fgvieira** [comment]
Same problem here as point `1`: reads with tandem repeats are kept. It would be great to have other methods (entropy or dust) to remove low complexity reads.

**Comment by miwipe** [comment]
I just wanted to echo the suggestions above. I think it could make fastp even more valuable to be able to filter low complex reads on the fly, using dust and or entropy.

**Comment by bpenaud** [comment]
Could be really nice to add this option.
